src/WCSimWCTriggerBase.cc

```diff
@@ -0,0 +1,270 @@
+#include "WCSimWCTriggerBase.hh"
+#include "WCSimWCPMT.hh"
+#include "WCSimWCDigi.hh"
+#include "WCSimWCHit.hh"
+#include "G4EventManager.hh"
+#include "G4Event.hh"
+#include "G4SDManager.hh"
+#include "G4DigiManager.hh"
+#include "G4ios.hh"
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+
+#include "WCSimDetectorConstruction.hh"
+#include "WCSimPmtInfo.hh"
+#include "WCSimDarkRateMessenger.hh"
+
+#include <vector>
+// for memset
+#include <cstring>
+#include <iostream>
+
+#ifndef WCSIMWCTRIGGERBASE_VERBOSE
+//#define WCSIMWCTRIGGERBASE_VERBOSE
+#endif
+
+const double WCSimWCTriggerBase::offset = 950.0 ; // ns. apply offset to the digit time
+const double WCSimWCTriggerBase::eventgateup = 950.0 ; // ns. save eventgateup ns after the trigger time
+const double WCSimWCTriggerBase::eventgatedown = -400.0 ; // ns. save eventgateup ns before the trigger time
+const double WCSimWCTriggerBase::LongTime = 100000.0 ; // ns = 0.1ms. event time
+
+
+WCSimWCTriggerBase::WCSimWCTriggerBase(G4String name,
+				       WCSimDetectorConstruction* myDetector,
+				       WCSimWCDAQMessenger* myMessenger)
+  :G4VDigitizerModule(name)
+{
+  G4String colName = "WCDigitizedCollection";
+  this->myDetector = myDetector;
```

Comment:
move this->myDetector = myDetector; to  myDetector(myDetector) in initializer list